File: apps/mac/Sources/ClipPlusMac/CoreBridge/CoreBridgeSmokeTest.swift
import Darwin
import Foundation

enum CoreBridgeSmokeTest {
    private static let expectedTestGroupId = "21YR2N3_wcdRPmEMLiuLMA"

    static func runIfRequested(environment: [String: String] = ProcessInfo.processInfo.environment) {
        guard environment["CLIPPLUS_COREBRIDGE_SMOKE_TEST"] == "1" else {
            return
        }

        let groupId = CoreBridge().deriveGroupId(for: "clipplus-test-key")
        print("corebridge_smoke_test group_id=\(groupId ?? "<nil>")")
        fflush(stdout)

        exit(groupId == expectedTestGroupId ? EXIT_SUCCESS : 2)
    }
}

File: apps/windows/ClipPlus.Windows/CoreBridge/CoreBridgeSmokeTest.cs
namespace ClipPlus.Windows.CoreBridge;

using System.IO;

internal static class CoreBridgeSmokeTest
{
    public static void RunIfRequested()
    {
        if (Environment.GetEnvironmentVariable("CLIPPLUS_COREBRIDGE_SMOKE_TEST") != "1")
        {
            return;
        }

        var coreBridge = new CoreBridge();
        var groupId = coreBridge.DeriveGroupId("clipplus-test-key");
        var output = groupId is null
            ? "corebridge_smoke_test failed"
            : $"corebridge_smoke_test group_id={groupId}";
        if (groupId is null)
        {
            output = $"{output}{Environment.NewLine}{string.Join(Environment.NewLine, coreBridge.FfiLoadDiagnostics())}";
        }

        var outputPath = Environment.GetEnvironmentVariable("CLIPPLUS_COREBRIDGE_SMOKE_TEST_OUTPUT");
        if (!string.IsNullOrWhiteSpace(outputPath))
        {
            File.WriteAllText(outputPath, output);
        }

        Environment.Exit(groupId == "21YR2N3_wcdRPmEMLiuLMA" ? 0 : 2);
    }
}
